<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ToolPaintWidget</class>
 <widget class="QWidget" name="ToolPaintWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>179</width>
    <height>122</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_preview">
       <property name="maximumSize">
        <size>
         <width>32</width>
         <height>32</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="combo_shapes">
       <item>
        <property name="text">
         <string>Rectangle</string>
        </property>
        <property name="icon">
         <iconset theme="draw-rectangle">
          <normaloff/>
         </iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Circle</string>
        </property>
        <property name="icon">
         <iconset theme="draw-circle">
          <normaloff/>
         </iconset>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Super Circle</string>
        </property>
        <property name="icon">
         <iconset theme="draw-circle"/>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Image</string>
        </property>
        <property name="icon">
         <iconset theme="image">
          <normaloff/>
         </iconset>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QStackedWidget" name="stacked_widget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page">
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="1">
        <widget class="QSpinBox" name="spin_width">
         <property name="suffix">
          <string> px</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1024</number>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Width</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="spin_height">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="suffix">
          <string> px</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1024</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Height</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2" rowspan="2">
        <widget class="QToolButton" name="check_ratio_locked">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Lock Ratio</string>
         </property>
         <property name="icon">
          <iconset theme="object-locked">
           <normaloff/>
          </iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_2">
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Diameter</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spin_diameter">
         <property name="suffix">
          <string> px</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1024</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_3">
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Diameter</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spin_super_diameter">
         <property name="suffix">
          <string> px</string>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1024</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>p-norm factor</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="spin_p_norm">
         <property name="minimum">
          <double>0.100000000000000</double>
         </property>
         <property name="maximum">
          <double>5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_4"/>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>check_ratio_locked</sender>
   <signal>toggled(bool)</signal>
   <receiver>spin_height</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>163</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>114</x>
     <y>96</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_width</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ToolPaintWidget</receiver>
   <slot>updateBrush()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>107</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>353</x>
     <y>57</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_height</sender>
   <signal>valueChanged(int)</signal>
   <receiver>ToolPaintWidget</receiver>
   <slot>updateBrush()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>93</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>combo_shapes</sender>
   <signal>activated(int)</signal>
   <receiver>ToolPaintWidget</receiver>
   <slot>updateBrush()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>134</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>296</x>
     <y>22</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>combo_shapes</sender>
   <signal>activated(int)</signal>
   <receiver>stacked_widget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>70</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_p_norm</sender>
   <signal>valueChanged(double)</signal>
   <receiver>ToolPaintWidget</receiver>
   <slot>updateBrush()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>138</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_super_diameter</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_width</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>125</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>92</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_width</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_diameter</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>92</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>55</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_diameter</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_width</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>157</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>105</x>
     <y>54</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spin_width</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spin_super_diameter</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>107</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>156</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateBrush()</slot>
 </slots>
</ui>
